docs/transpile/defaults_and_configuration_options.mdx

```diff
@@ -1,5 +1,415 @@
+
+**Layout stage**
+
+- initial_layout (Layout | Dict | List) - Initial position of virtual qubits on physical qubits. If this layout makes the circuit compatible with the `coupling_map` constraints, it will be used. The final layout is not guaranteed to be the same, as the transpiler might permute qubits through swaps or other means. For full details, see the [initial_layout section.](#initial_layout)
+- layout_method (str) - Name of layout selection pass (‘trivial’, ‘dense’, ‘noise_adaptive’, or ‘sabre’). This can also be the external plugin name to use for the layout stage. You can see a list of installed plugins by running `list_stage_plugins()` with `layout` for the `stage_name` argument.
+
+**Routing stage**
+- routing_method (str) - Name of routing pass (‘basic’, ‘lookahead’, ‘stochastic’, ‘sabre’, or  ‘none’). This can also be the external plugin name to use for the routing stage. You can see a list of installed plugins by running `list_stage_plugins()` with `routing` for the `stage_name` argument.
+- seed_transpiler (int) - Sets random seeds for the stochastic parts of the transpiler.
+
+**Optimization stage**
+- approximation_degree (float | None) - Heuristic dial used for circuit approximation (1.0 = no approximation, 0.0 = maximal approximation). The default value is 1.0.  Specifying `None` sets the approximation degree to the reported error rate. See the [Approximation degree section](#approximation-degree) for more details.
+- optimization_level (int) - How much optimization to perform on the circuits. Integer in the range (0 - 3). Higher levels generate more optimized circuits, at the expense of longer transpilation time. The default is `1`. See the [Optimization level section](#optimization-level) for more details.
+- optimization_method (str) - The plugin name to use for the optimization stage. By default an external plugin is not used. You can see a list of installed plugins by running `list_stage_plugins()` with `optimization` for the `stage_name` argument.
+
+
+**Scheduling stage**
+- scheduling_method (str) - Name of the scheduling pass. This can also be the external plugin name to use for the scheduling stage. You can see a list of installed plugins by running `list_stage_plugins()` with `scheduling` for the `stage_name` argument.
+  * 'as_soon_as_possible': Schedule instructions greedily, as early as possible on a qubit resource. (alias: 'asap') 
+  * 'as_late_as_possible': Schedule instructions late, that is, keeping qubits in the ground state when possible. (alias: 'alap') 
+
+**UNKNOWN**
+
+- callback (Callable[[BasePass, DAGCircuit, float, PropertySet, int], Any]) - A callback function that is called after each pass execution. The function is called with these keyword arguments:
```

Comment:
I think this callback argument could use a little more explanation, as a new user i don't really understand what this is for